api/src/test/java/engineering/everest/lhotse/api/rest/ExceptionHandlingControllerAdviceTest.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    @Test
+    void willMapExecutionExceptions_WhenCauseIsAnIllegalArgumentException() {
+        var exception = new ExecutionException("outer exception", new IllegalArgumentException("bad argument"));
+        var expectedResponse = ApiErrorResponse.builder()
+                .status(BAD_REQUEST)
+                .message("bad argument")
+                .timeStamp(Instant.now(clock))
+                .build();
+
+        assertEquals(new ResponseEntity<>(expectedResponse, new HttpHeaders(), BAD_REQUEST), controllerAdvice.handleExecutionException(exception));
+    }
+
+    @Test
+    void annotatedWithExceptionHandlerForRemoteCommandExecutionException() throws NoSuchMethodException {
+        Method handleRemoteCommandExecutionException = controllerAdvice.getClass().getMethod("handleRemoteCommandExecutionException", RemoteCommandExecutionException.class);
+        assertDoesNotThrow(() -> handleRemoteCommandExecutionException.getAnnotation(ExceptionHandler.class));
+    }
+
+    @Test
+    void willMapRemoteCommandExecutionExceptionWithNestedExecutionException() {
+        var exception = new RemoteCommandExecutionException(new ExecutionException("inner exception", new IllegalArgumentException("bad argument")));
+        var expectedResponse = ApiErrorResponse.builder()
+                .status(BAD_REQUEST)
+                .message("bad argument")
+                .timeStamp(Instant.now(clock))
+                .build();
+
+        assertEquals(new ResponseEntity<>(expectedResponse, new HttpHeaders(), BAD_REQUEST), controllerAdvice.handleRemoteCommandExecutionException(exception));
+    }
+
+    @Test
+    void willMapRemoteCommandExecutionExceptionWithNestedInterruptedException() {
+        var exception = new RemoteCommandExecutionException(new InterruptedException("inner exception"));
+        var expectedResponse = ApiErrorResponse.builder()
+                .status(BAD_REQUEST)
+                .message("inner exception")
+                .timeStamp(Instant.now(clock))
+                .build();
+
+        assertEquals(new ResponseEntity<>(expectedResponse, new HttpHeaders(), BAD_REQUEST), controllerAdvice.handleRemoteCommandExecutionException(exception));
+    }
+
+    @Test
+    void willMapRemoteCommandExecutionExceptionWithNestedException() {
+        var exception = new RemoteCommandExecutionException(new Exception("inner exception"));
+        var expectedResponse = ApiErrorResponse.builder()
+                .status(BAD_REQUEST)
+                .message("inner exception")
+                .timeStamp(Instant.now(clock))
+                .build();
+
+        assertEquals(new ResponseEntity<>(expectedResponse, new HttpHeaders(), BAD_REQUEST), controllerAdvice.handleRemoteCommandExecutionException(exception));
+    }
+
+    @Test
+    void willMapMethodArgumentNotValidExceptions() throws Exception {
+        var bindingResult = mock(BeanPropertyBindingResult.class);
+        var errors = List.of(
```

Comment:
Static import.

Reply:
Not in this case; List.of() is readable, of() is semantically ... lost.